### ISSUE TITLE
Build fails with Instructions

### ISSUE THREAD

**Issue by jtmonroe** [fabianmurariu/graphblas-java-native#15]
The build instructions as listed in the readme fail on my device. After cloning recursively, I run the command `make graphblas-java` from the parent directory and receive the following message.

`make: *** No rule to make target 'graphblas-java'.  Stop.`

I attempt to run the same command inside the repository and receive the message

`make: Nothing to be done for 'graphblas-java'.`

So instead I simply run `make` within the directory. This produces the three `.so` files (`libgraphblas.so`, `libgraphblas.so.4`, and `libgraphblas.so.4.0.0`). However, I receive a build error as

```
 130 ❯ make
LD_LIBRARY_PATH=/home/joelm/Code/graphblas-java-native/GraphBLAS/build:$LD_LIBRARY_PATH mvn clean install
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.inject.internal.cglib.core.$ReflectUtils$1 (file:/usr/share/maven/lib/guice.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)
WARNING: Please consider reporting this to the maintainers of com.google.inject.internal.cglib.core.$ReflectUtils$1
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[INFO] Scanning for projects...
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.github.fabianmurariu:graphblas-java:nar:0.1.24
[WARNING] 'dependencies.dependency.(groupId:artifactId:type:classifier)' must be unique: org.scijava:native-lib-loader:jar -> version 2.3.4 vs 2.2.0 @ line 118, column 21
[WARNING] 'build.plugins.plugin.version' for org.codehaus.mojo:build-helper-maven-plugin is missing. @ line 392, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 285, column 21
[WARNING] 'build.plugins.plugin.version' for com.github.maven-nar:nar-maven-plugin is missing. @ line 293, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-deploy-plugin is missing. @ com.github.fabianmurariu:graphblas-java-native:0.1.24, /home/joelm/Code/graphblas-java-native/pom.xml, line 155, column 21
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.github.fabianmurariu:graphblas-package:jar:0.1.24
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ com.github.fabianmurariu:graphblas-java-native:0.1.24, /home/joelm/Code/graphblas-java-native/pom.xml, line 110, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-shade-plugin is missing. @ line 37, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-deploy-plugin is missing. @ com.github.fabianmurariu:graphblas-java-native:0.1.24, /home/joelm/Code/graphblas-java-native/pom.xml, line 155, column 21
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] Some problems were encountered while building the effective model for com.github.fabianmurariu:graphblas-java-native:pom:0.1.24
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-compiler-plugin is missing. @ line 110, column 21
[WARNING] 'build.plugins.plugin.version' for org.apache.maven.plugins:maven-deploy-plugin is missing. @ line 155, column 21
[WARNING] The expression ${version} is deprecated. Please use ${project.version} instead.
[WARNING] 
[WARNING] It is highly recommended to fix these problems because they threaten the stability of your build.
[WARNING] 
[WARNING] For this reason, future Maven versions might no longer support building such malformed projects.
[WARNING] 
[INFO] Inspecting build with total of 3 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 3 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO] ------------------------------------------------------------------------
[INFO] Detecting the operating system and CPU architecture
[INFO] ------------------------------------------------------------------------
[INFO] os.detected.name: linux
[INFO] os.detected.arch: x86_64
[INFO] os.detected.version: 5.11
[INFO] os.detected.version.major: 5
[INFO] os.detected.version.minor: 11
[INFO] os.detected.release: ubuntu
[INFO] os.detected.release.version: 20.04
[INFO] os.detected.release.like.ubuntu: true
[INFO] os.detected.release.like.debian: true
[INFO] os.detected.classifier: linux-x86_64
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO] 
[INFO] GraphBLAS JNI Bindings Parent                                      [pom]
[INFO] GraphBLAS JNI Bindings                                             [nar]
[INFO] graphblas-package                                                  [jar]
[INFO] 
[INFO] -----------< com.github.fabianmurariu:graphblas-java-native >-----------
[INFO] Building GraphBLAS JNI Bindings Parent 0.1.24                      [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO] 
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ graphblas-java-native ---
[INFO] Deleting /home/joelm/Code/graphblas-java-native/target
[INFO] 
[INFO] --- maven-javadoc-plugin:3.2.0:jar (javadoc-jar) @ graphblas-java-native ---
[INFO] Not executing Javadoc as the project is not a Java classpath-capable package
[INFO] 
[INFO] --- maven-gpg-plugin:1.5:sign (sign-artifacts) @ graphblas-java-native ---
gpg: no default secret key: No secret key
gpg: signing failed: No secret key
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for GraphBLAS JNI Bindings Parent 0.1.24:
[INFO] 
[INFO] GraphBLAS JNI Bindings Parent ...................... FAILURE [  0.528 s]
[INFO] GraphBLAS JNI Bindings ............................. SKIPPED
[INFO] graphblas-package .................................. SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  1.492 s
[INFO] Finished at: 2021-09-03T12:27:19-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-gpg-plugin:1.5:sign (sign-artifacts) on project graphblas-java-native: Exit code: 2 -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
make: *** [Makefile:20: build] Error 1
```

After this, since the `.so` files are still being generated, I attempted to reference them directly using `System.load(${PATH_TO_SO_FILES})` in my java code for each of the `.so files`. This produced the error message seen below. 

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f43a9ce19f6, pid=28719, tid=28720
#
# JRE version: OpenJDK Runtime Environment Corretto-11.0.12.7.1 (11.0.12+7) (build 11.0.12+7-LTS)
# Java VM: OpenJDK 64-Bit Server VM Corretto-11.0.12.7.1 (11.0.12+7-LTS, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0xf109f6]  UTF8::unicode_length(char const*, bool&, bool&)+0x6
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/share/apport/apport %p %s %c %d %P %E" (or dumping to /home/joelm/Code/factweave-code/core.28719)
#
# An error report file with more information is saved as:
# REDACTED PATH
#
# If you would like to submit a bug report, please visit:
#   https://github.com/corretto/corretto-11/issues/
#
Error in GRB: 13 

Process finished with exit code 134 (interrupted by signal 6: SIGABRT)
```

In addition to your code. I have attempted to use the Maven [release](https://mvnrepository.com/artifact/com.github.fabianmurariu/graphblas-java-native) with the newest version (0.1.18). This is what began the issue. If I run the command `    GRBCORE.createMatrix(GRAPHBLAS.intType(), 100L, 100L)`  without the `System.load` commands, I receive the message (with stack trace)

```
Exception in thread "main" java.lang.UnsatisfiedLinkError: /tmp/libgraphblas-java-0.1.1817759710730681086144.so: libgraphblas.so.4: cannot open shared object file: No such file or directory
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
	at java.base/java.lang.Runtime.load0(Runtime.java:768)
	at java.base/java.lang.System.load(System.java:1837)
	at com.github.fabianmurariu.unsafe.NarSystem.loadLibrary(NarSystem.java:48)
	at com.github.fabianmurariu.unsafe.GRAPHBLAS.<clinit>(GRAPHBLAS.java:9)
	at com.ng2.mgr.index.structs.tensor.matrix.GraphBlasMatrixKt.main(GraphBlasMatrix.kt:50)
	at com.ng2.mgr.index.structs.tensor.matrix.GraphBlasMatrixKt.main(GraphBlasMatrix.kt)
```
I noticed that the command `make graphblas-java` seems out of date, as I can find no references to it within the makefile. What steps forward should I take to integrate your JNI code to my codebase?

**Comment by jtmonroe** [comment]
I successfully got the code to both work, and to run in Kotlin. I got the build to run by removing the artifact signing on line 20 of the Makefile. I added `-Dgpg.skip` and this allowed the build to complete successfully. Secondly, I set the `LD_LIBRARY_PATH` to the directory containing all of the relevant `.so` files. Lastly, after consulting the API, I ran `GRBCORE.initNonBlocking()` before using the code.
